### PR TITLE
Simplify assembly load on CoreCLR

### LINKED ISSUE #4328: Simplify assembly load on CoreCLR
Just use the default load context for razor views. We get no benefit from having a custom load context. Now that we can load things directly into the default load context, we don't need it.

/cc @pranavkm

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #4328

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl \ @NTaylorMullen

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -260,24 +261,27 @@ private List<MetadataReference> GetApplicationReferences()
                     continue;
                 }
 
-                metadataReferences.AddRange(referencePaths.Select(CreateMetadataFileReference));
+                foreach (var path in referencePaths)
+                {
+                    if (libraryPaths.Add(path))
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: This a separate issue?

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 2]
```diff
@@ -247,6 +247,7 @@ private List<MetadataReference> GetApplicationReferences()
                 return metadataReferences;
             }
 
+            var libraryPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Can you file an issue to make sure we don't hold onto the dependency context forever?

**Pull request closed without merging**